tests/system/libraries/ChromeLib.py

```diff
@@ -0,0 +1,134 @@
+from robot.libraries.Process import Process as _ProcessLib
+from KeyInputLib import KeyInputLib as _KeyInputLib
+from AssertsLib import AssertsLib as _AssertsLib
+import NvdaLib as _NvdaLib
+
+builtIn: BuiltIn = BuiltIn()
+opSys: _OpSysLib = _getLib('OperatingSystem')
+process: _ProcessLib = _getLib('Process')
+keyInputLib: _KeyInputLib = _getLib('KeyInputLib')
+assertsLib: _AssertsLib = _getLib('AssertsLib')
+
+
+# In Robot libraries, class name must match the name of the module. Use caps for both.
+class ChromeLib:
+	_testFileStagingPath = _tempfile.mkdtemp()
+
+	def __init__(self):
+		self.chromeHandle: _Optional[int] = None
+
+	@staticmethod
+	def _getTestCasePath(filename):
+		return _pJoin(ChromeLib._testFileStagingPath, filename)
+
+	def start_chrome(self, filePath):
+		builtIn.log(f"starting chrome: {filePath}")
+		self.chromeHandle = process.start_process(
+			f"start chrome \"{filePath}\"",
+			shell=True,
+			alias='chromeAlias',
+		)
+		process.process_should_be_running(self.chromeHandle)
+		return self.chromeHandle
+
+	_testCaseTitle = "NVDA Browser Test Case"
+	_beforeMarker = "Before Test Case Marker"
+	_afterMarker = "After Test Case Marker"
+
+	@staticmethod
+	def _writeTestFile(testCase) -> str:
+		"""
+		Creates a file for a HTML test case. The sample is written with a button before and after so that NVDA
+		can tab to the sample from either direction.
+		@param testCase:  The HTML sample that is to be tested.
+		@return: path to the HTML file.
+		"""
+		filePath = ChromeLib._getTestCasePath("test.html")
+		fileContents = (f"""
+			<head>
+				<title>{ChromeLib._testCaseTitle}</title>
+			</head>
+			<body>
+				<button>{ChromeLib._beforeMarker}</button>
+				{testCase}
+				<button>{ChromeLib._afterMarker}</button>
+			</body>
+		""")
+		with open(file=filePath, mode='w', encoding='UTF-8') as f:
+			f.write(fileContents)
+		return filePath
+
+	def prepareChrome(self, testCase: str) -> None:
+		"""
+		Starts Chrome opening a file containing the HTML sample
+		@param testCase - The HTML sample to test.
+		"""
+		spy = _NvdaLib.getSpyLib()
+		path = self._writeTestFile(testCase)
+		self.start_chrome(path)
+		# Ensure chrome started
+		applicationTitle = f"{self._testCaseTitle}"
+		spy.wait_for_specific_speech(applicationTitle)
+		# Different versions of chrome have small variations in the separator between document name and
+		# application name. E.G. "htmlTest   Google Chrome", "html – Google Chrome"
+		# Rather than try to get the separator right, wait for the doc title, then wait for the 'Google Chrome'
+		# part. Note: this also works with "Chrome Canary"
+		spy.wait_for_specific_speech("Google Chrome")
+		# Read all is configured, but just test interacting with the sample.
+		spy.wait_for_speech_to_finish()
+
+		# move to start marker
+		for i in range(10):  # set a limit on the number of tries.
+			# Small changes in Chrome mean the number of tab presses to get into the document can vary.

```

Comment:
have you considered pressing f6 to get to the document area? May be that's more reliable?

Reply:
I hadn't considered that. For now I'll leave it as it is, but it's worth keeping in mind.